azurerm/helpers/azure/ssh_key.go

```diff
@@ -0,0 +1,28 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+// NormaliseSSHKey attempts to remove invalid formatting and line breaks that can be present in some cases
+// when querying the Azure APIs
+func NormaliseSSHKey(input string) (*string, error) {
+	if input == "" {
+		return nil, fmt.Errorf("empty string supplied")
+	}
+
+	// Strip the Azure generated multiline wrapper
+	Normalised := strings.ReplaceAll(input, "<<~EOT\r\n", "")
+	Normalised = strings.ReplaceAll(Normalised, "EOT", "")
+
+	// strip Windows flavour new-lines
+	Normalised = strings.ReplaceAll(Normalised, "\r\n", "")
+
+	// strip Linux flavour nee lines
+	Normalised = strings.ReplaceAll(Normalised, "\n", "")
+
+	return utils.String(Normalised), nil
```

Comment:
this won't handle whitespace surrounding each line of the key (which is possible when the key is nested in a config) - ultimately this'll need to become:

```suggestion
	output := input
	output = strings.ReplaceAll(output, "<<~EOT", "")
	output = strings.ReplaceAll(output, "EOT", "")
	output = strings.ReplaceAll(output, "\r", "")

	lines := make([]string, 0)
	for _, line := range strings.Split(output, "\n") {
		lines = append(lines, strings.TrimSpace(line))
	}
	normalized := strings.Join(lines, "")
	return utils.String(normalized), nil
```

Reply:
It's not possible to specify a key like this in config, it would fail validation. This covers the case where a VM is imported and the Azure generated public key is line wrapped from the ARM response in the state. afaik, there's no other way this would be needed?